jax/experimental/jax2tf/jax2tf.py

```diff
@@ -41,10 +41,38 @@
 from tensorflow.compiler.tf2xla.python import xla as tfxla  # type: ignore[import]
 from tensorflow.compiler.xla import xla_data_pb2  # type: ignore[import]
 
-# A value suitable in a TF tracing context: tf.Tensor, tf.Var, or
-# Python scalar or numpy.ndarray.
+# A value suitable in a TF tracing context: tf.Tensor, tf.Variable, tf.EagerTensor,
+# or Python scalar or numpy.ndarray.
 TfVal = Any
 
+# During JAX transformations we sometimes produce a Jaxpr that has arguments
+# of abstract value core.abstract_unit and results equal to core.unit.
+# These are arguments and results that are not used in the computation.
+# Whenever we are in a JAX tracing context we must use `core.unit` values
+# in those places. However, when we move to TF we have to turn them into
+# some small TFVal; it does not matter which value since it will never be used
+# in an actual operation.
+TfValOrUnit = Union[TfVal, core.Unit]
+
+_unit_tfval = tf.constant(9, tf.uint8)
```

Comment:
Perhaps f32 `NaN`? Incase we leak this in somewhere it would be better to propagate `NaN`s not scale/shift by 9.

Reply:
I like the idea of NaN. I think it is pretty unlikely that this will leak, but nan is safer.